vars/setupCommonPipelineEnvironment.groovy

```diff
@@ -2,13 +2,38 @@ def call(Map parameters = [:]) {
 
     handlePipelineStepErrors (stepName: 'setupCommonPipelineEnvironment', stepParameters: parameters) {
 
-        def configFile = parameters.get('configFile', '.pipeline/config.properties')
-        def script = parameters.script
+        prepareDefaultValues script: script
 
-        Map configMap = [:]
-        if (configFile.length() > 0)
-            configMap = readProperties (file: configFile)
+        String configFile = parameters.get('configFile')
+
+        loadConfigurationFromFile(configFile)
+    }
+}
+
+private boolean isYaml(String fileName) {
+    return fileName.endsWith(".yml") || fileName.endsWith(".yaml")
+}
+
+private boolean isProperties(String fileName) {
+    return fileName.endsWith(".properties")
+}
+
+private loadConfigurationFromFile(script, String configFile) {
+
+    String defaultPropertiesYmlConfigFile = '.pipeline/config.properties'
+    String defaultYmlConfigFile = 'pipeline_config.yml'
```

Comment:
can we move the default Config file also to .pipeline folder?

Reply:
I am still in favor of having it in the root as the manifest.yml usually is also in the root and has a similar meaning, same with package.json and all the other configuration files for the tools.
Furthermore I see the problem that folders with dots in the beginning can be inconvenient for the user. For example:
* I personally have the feeling that a folder with dot does not belong to my project and should not be committed to git.
* On Windows in the explorer I cannot create a folder with a dot.
* On Linux/Mac these folders are treated as hidden folders. However, we see the configuration as essential part of the project and thus should be visible.

Reply:
What else does the `.pipeline` directory contain?

I think it is fair to compare the pipeline configuration to a `Makefile` or a `pom.xml`. Those files are not dotfiles, because they are a part of the software delivery. They describe how the software is built. The same holds true for the pipeline configuration, in my opinion.

I think about it this way: If I had a "export this source code repository to a zip file", what would I expect this repository to contain? Maybe not the `.gitigonore` file, because that is just an auxiliary file for git usage. Also maybe not a `.editorconfig` file, because this is just an auxiliary file. But surely, `Makefile`, `pom.xml`, `Jenkinsfile` and `pipeline_config.yml` needs to be in there, because they describe how the project is built, they *are* (part of) the software delivery.

I'm curious to know the rationale for `.pipeline`. If it only contains one file, the question is why this needs to be a directory?

Reply:
> I personally have the feeling that a folder with dot does not belong to my project and should not be committed to git.

IMO, we should not base our decisions on personal feelings. Files that should not be committed to Git are to be mentioned in a `.gitignore` file. There is no convention on the file name for this.

> On Windows in the explorer I cannot create a folder with a dot.

This is not true. Try creating a folder named `.pipelines.`. Windows will remove the trailing dot and you will have the expected outcome. I'm not saying that this is a good way of handling folder names starting with a dot, but it is an issue with Windows Explorer and should be fixed there.

> If I had a "export this source code repository to a zip file", what would I expect this repository to contain?

I would definitely not expect it to contain any configurations of deployment hosts. URLs of test systems could be strictly internal. In some companies leaking internal URLs is considered to be a security issue and must not happen.

> Maybe not the .gitigonore file, because that is just an auxiliary file for git usage

So why is `.gitignore` an auxiliary file for git usage but `Jenkinsfile` and `pipeline_config.yml` are not an auxiliary files for Jenkins usage? It's not like there are no alternatives to Jenkins. There are people who orchestrate using e.g. concourse.ci. One can switch that layer at all times and still build the same software, just like if they switched from Git to Mercurial.

Reply:
> I'm curious to know the rationale for .pipeline. If it only contains one file, the question is why this needs to be a directory?

Here I agree. In case we do not expect other files a .pipeline file would be sufficient.

Regarding the general question about dot files:
Some prominent examples on linux like systems are:
* ```.bash_profile``` contains settings for the bash
* ```.m2``` contains maven settings file, credentials with encrypted passwds and local repo 
* ```.mozilla``` Settings and cache for the firefox
* ```.kde```, a lot of settings etc. for KDE (in case KDE is used)
* ```.npm``` npm configurations
* ```.npmrc``` a cache
* ```.netrc``` credentials (can be used e.g. with curl)
* ```.ssh``` ssh settings, known hosts, private keys
* ```.gitignore``` (was already mentiones)

Now we should ask ourselfs: does ```.pipeline``` fits into that pattern?

Regarding ```pom.xml ```(controlling the build) is not a dot file: It is not a dot file, but ```.m2``` (containing among other things some fundamental maven settings (more the less build server specific)) is a dot file.

That relates IMO well to ```Jenkinsfile``` vs ```.pipeline```.

I personally think the semantics of our ```.pipeline``` file fits into the dotfile examples provided above.

Reply:
There is also a reasoning for the folder ...

Teams can store additional files in there. We for example have team-specific pipeline extensions in there. They may also add other custom files there, additional config etc.
And that is what teams do today ...
Why should we change it for them?

Reply:
Okay. I think one reason for the discussion is that in our projects the configuration file is used in a different way. Since, it is still configurable over the parameters, I suggest that we keep the folder .pipeline. I already changed the step loading it.